code/modules/research/xenobiology/crossbreeding/gentle.dm

```diff
@@ -21,18 +21,21 @@
 	extract.forceMove(src)
 
 /obj/item/slimecross/gentle/attack_self(mob/living/carbon/user)
-	if(user.incapacitated() || !iscarbon(user))
-		return
-	if(!COOLDOWN_FINISHED(src, use_cooldown))
-		return
-	COOLDOWN_START(src, use_cooldown, extract.activate(user, user.dna.species, SLIME_ACTIVATE_MINOR))
+	if(activate_core(user))
+		COOLDOWN_START(src, use_cooldown, extract.activate(user, user.dna.species, SLIME_ACTIVATE_MINOR))
 
 /obj/item/slimecross/gentle/AltClick(mob/living/carbon/user, obj/item/I)
+	if(activate_core(user))
+		COOLDOWN_START(src, use_cooldown, extract.activate(user, user.dna.species, SLIME_ACTIVATE_MAJOR))
+
+/obj/item/slimecross/gentle/proc/activate_core(mob/living/carbon/user)
```

Comment:
I don't like the proc name. the proc's purpose is not activate. it's checking if you can use it.

Reply:
adjusted the name